### DISCUSSION TITLE
Question about screen scaling.

### DISCUSSION THREAD

**Discussion by achrystie** [nesbox/TIC-80#1603] [Q&A]
This is my first post, and it's not a complaint, just trying to get a handle on what I can do with scaling for hardware as I'm experimenting with the concept of creating my own arcade machine(s), non commercial, that have properly pixel perfect scaled classic style games, clones of a sort, but designed for wide aspect ratios and have their own dedicated hardware/cabinet.
Right now on a 1080p monitor the 136 vertical resolution scales just over 1080p (1088) if the horizontal was scaled to 1920 (8x240).
Obviously this doesn't pixel scale properly so when in full screen there are bars/a background around the game window, and I think it's cool that the color of the background can be customized, but I'd prefer to minimize the effect of this.  With that said, if I was to use a 16:10 monitor, would the screen scale such that the horizontal resolution filled the screen and the vertical showed the bars only at the top and bottom?  I'd test it, but I don't have a 16:10 monitor on hand atm.
For example a 1440x900 monitor, I would think, should scale such that the horizontal is 6x the size (6 x 240 = 1440) and the vertical is scaled to (136 x 6 = 816), leaving 42 pixels on top and bottom. is that the case, or does it always undershoot the resolution when in full screen such that there is a background/bars on all four sides?
Second question, is there a way to interface with GPIO pins for something like a raspberry PI in TIC-80, similar to how it can be done in Pico 8? 
Anyway I hope this doesn't come off as me being negative about the software, I love it so far, and have some hopes about the wider screen resolution providing a bit more gameplay options than Pico 8, which I also like. but have different plans for, so I figured I'd ask.

**Comment by ChildishGiant** [comment]
I'm pretty certain that the scaling is pixel perfect like you've described. As for GPIO there's no current functionality that I'm aware of, although feel free to open an issue for it.

**Reply by achrystie** [reply]
Thank you for the reply.  I may buy a cheap refurbished 16:10 monitor to test it out.  I'll update if I find it doesn't scale as expected.

**Comment by achrystie** [comment]
Sorry to necro my own post, but I obtained (found a monitor from an old computer my mother had, with a 16:10 aspect ratio), and the scaling does NOT scale 6x to fill the horizontal resolution and put bars only on top and bottom.  The monitor is 1440 x 900, 17", and I still get bars on all four sides.  Is there some sort of configuration setting(s) I can adjust to try to force it to scale horizontally to 1440 and vertically to 816 with bars only on top and bottom?

**Comment by nesbox** [comment]
The TIC-80 has a screen resolution of 256x144 with a 16:9 aspect ratio where the drawing area is 240x136.
Unfortunately, there is no way to adjust it in the settings, sorry.
The only way I see is to make your own build, you can go to [`include/tic80.h`](https://github.com/nesbox/TIC-80/blob/main/include/tic80.h) change `TIC80_FULLWIDTH / TIC80_FULLHEIGHT` and build.

``` c
#define TIC80_FULLWIDTH         240
#define TIC80_FULLHEIGHT        136
```

**Reply by achrystie** [reply]
Thank you for the reply. I'll see about building it from source.

**Reply by joshgoebel** [reply]
Just to be clear changing this probably isn't the best way to fix the black bars...

**Comment by bennypowers** [comment]
Here is a screenshot of tic80 (flatpak) running on fedora workstation 39, in a GNOME wayland session, with GNOME fractional scaling enabled and UI scaling set to 125%
![image](https://github.com/nesbox/TIC-80/assets/1466420/f262996e-61cd-4317-9a90-873de1a9267d)
and here's a screenshot with UI scaling set to 200% (integer scaling)
![image](https://github.com/nesbox/TIC-80/assets/1466420/dbacc1cd-10bc-44c4-80b0-43fbf9daf81c)
and with scaling disabled (100%)
![image](https://github.com/nesbox/TIC-80/assets/1466420/8977eccd-a62b-4043-9446-0baa26cecb5e)

Please note that resizing the window has no effect, as the scale of the rendered fantasy console screen relative to the window size remains constant.

**Comment by Yiannis128** [comment]
> Here is a screenshot of tic80 (flatpak) running on fedora workstation 39, in a GNOME wayland session, with GNOME fractional scaling enabled and UI scaling set to 125% ![image](https://private-user-images.githubusercontent.com/1466420/286150349-f262996e-61cd-4317-9a90-873de1a9267d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NDYyNjIsIm5iZiI6MTcxMzc0NTk2MiwicGF0aCI6Ii8xNDY2NDIwLzI4NjE1MDM0OS1mMjYyOTk2ZS02MWNkLTQzMTctOWE5MC04NzNkZTFhOTI2N2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjJUMDAzMjQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTkwY2MwZWRlMmJmZjhhZThmZTlkNDFlYzJlYTMwZjFmMDk1ZGMzNzIwN2Q0YTY2NzVjYjVhNmM1M2ZjZmE4NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.4M68lfu0ZZGrlamPpGykynI8DMquyuzAQXmnmTHyFbk) and here's a screenshot with UI scaling set to 200% (integer scaling) ![image](https://private-user-images.githubusercontent.com/1466420/286150577-dbacc1cd-10bc-44c4-80b0-43fbf9daf81c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NDYyNjIsIm5iZiI6MTcxMzc0NTk2MiwicGF0aCI6Ii8xNDY2NDIwLzI4NjE1MDU3Ny1kYmFjYzFjZC0xMGJjLTQ0YzQtODBiMC00M2ZiZjlkYWY4MWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjJUMDAzMjQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWZmOWY4OTJlYzZlM2FkNWMyMjkyZjJiYWZjMmQ1NzdkNmY3YzUyNDhkZjc2NWViZDJjM2UwNGIwNTc5ODc3NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ptlFcjsKMK-2MnolJw6rFp1tDpO0_81CLJYJLKJ3iNQ) and with scaling disabled (100%) ![image](https://private-user-images.githubusercontent.com/1466420/286150676-8977eccd-a62b-4043-9446-0baa26cecb5e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NDYyNjIsIm5iZiI6MTcxMzc0NTk2MiwicGF0aCI6Ii8xNDY2NDIwLzI4NjE1MDY3Ni04OTc3ZWNjZC1hNjJiLTQwNDMtOTQ0Ni0wYmFhMjZjZWNiNWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjJUMDAzMjQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGQwMzJiNWQyOTg0Zjg0ODA5YTMzNmE0NzEzMmM2M2U4N2ZkYTMzNGRmYTA3M2NiMTM1OTU3YTQ3ZWI4ZGQ0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.eB9MDtCAYi4WzmNqjJ2YAK86VBaZ69pBl-7W2EecYbo)
> 
> Please note that resizing the window has no effect, as the scale of the rendered fantasy console screen relative to the window size remains constant.

I'm also experiencing an issue like this, is there a way around it? I'm on Wayland Gnome running the Flatpak as well...
